growth/too/models.py

```diff
@@ -945,6 +947,164 @@ class Observation(db.Model):
+        tot = 0
+        observation_ids = []
+        for ii, observation in enumerate(observations):
+            if observation.observation_id not in observation_ids:
+                tot = tot + observation.exposure_time/60.0
+                observation_ids.append(observation.observation_id)
+        return tot
+
+    def limmag(self, filt, start_time, end_time):
+        observations = self.get_observations(filt,
+                                             start_time, end_time)
+        limmags = []
+        for ii, observation in enumerate(observations):
+            if observation.limmag is not None:
+                limmags.append(observation.limmag)
+        return np.median(limmags)
+
+    def get_probability(self, filt, start_time, end_time):
+        localization = Localization.query.filter_by(
+            dateobs=self.dateobs,
+            localization_name=self.localization_name).one()
+
+        ipix = np.asarray(list(self.ipix(filt, start_time, end_time)))
+        if len(ipix) > 0:
+            return localization.flat_2d[ipix].sum()
+        else:
+            return 0.0
+
+    def get_observations(self, filt, start_time, end_time):
+
+        localization = Localization.query.filter_by(
+            dateobs=self.dateobs,
+            localization_name=self.localization_name).one()
+        prob = localization.flat_2d
+
+        fields = Field.query.filter_by(telescope=self.telescope).all()
+        field_prob_1, field_ids_1 = [], []
+        field_prob_2, field_ids_2 = [], []
+        for field in fields:
+            if field.field_id < 800:
+                field_ids_1.append(field.field_id)
+                field_prob_1.append(np.sum(prob[field.ipix]))
+            else:
+                field_ids_2.append(field.field_id)
+                field_prob_2.append(np.sum(prob[field.ipix]))
+        field_ids_1, field_ids_2 = np.array(field_ids_1), np.array(field_ids_2)
+        field_prob_1 = np.array(field_prob_1)
+        field_prob_2 = np.array(field_prob_2)
+
+        field_ids_1 = field_ids_1[np.argsort(field_prob_1)[::-1]]
+        field_prob_1 = np.sort(field_prob_1)[::-1]
+
+        field_ids_2 = field_ids_2[np.argsort(field_prob_2)[::-1]]
+        field_prob_2 = np.sort(field_prob_2)[::-1]
+
+        cumsum = np.cumsum(field_prob_1)
+        idx2 = np.argmin(np.abs(cumsum-0.99))
+        field_ids_1 = field_ids_1[:idx2]
+        field_prob_1 = field_prob_1[:idx2]
+
+        cumsum = np.cumsum(field_prob_2)
+        idx2 = np.argmin(np.abs(cumsum-0.99))
+        field_ids_2 = field_ids_2[:idx2]
+        field_prob_2 = field_prob_2[:idx2]
+
+        field_ids = np.hstack((field_ids_1,field_ids_2))
+
+        telescope = Telescope.query.filter_by(telescope=self.telescope).one()
+        filts = list(telescope.filters)
+        filts.append('a')
+
+        bands = {'g': 1, 'r': 2, 'i': 3, 'z': 4, 'J': 5, 'U': 6, 'a': 7}
+        filter_id = bands[filt]
+        if filt == "a":
```

Comment:
What is an 'a' filter?

Reply:
This should retrieve the observations in all filters, which is displayed in the last row of the table.